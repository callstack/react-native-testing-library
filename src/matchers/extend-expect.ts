/// <reference path="./extend-expect.d.ts" />

import { toBeOnTheScreen } from './to-be-on-the-screen';
import { toBeEmptyElement } from './to-be-empty-element';
import { toBeVisible } from './to-be-visible';
import { toHaveDisplayValue } from './to-have-display-value';
import { toHaveTextContent } from './to-have-text-content';

expect.extend({
  toBeOnTheScreen,
  toBeEmptyElement,
  toBeVisible,
  toHaveDisplayValue,
  toHaveTextContent,
});
