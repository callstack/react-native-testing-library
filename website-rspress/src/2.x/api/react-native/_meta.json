[
  {
    "type": "file",
    "name": "index",
    "label": "Table of Contents"
  },
  {
    "type": "dir",
    "name": "classes",
    "label": "Classes"
  },
  {
    "type": "dir",
    "name": "interfaces",
    "label": "Interfaces"
  },
  {
    "type": "dir",
    "name": "types",
    "label": "Types"
  }
]
