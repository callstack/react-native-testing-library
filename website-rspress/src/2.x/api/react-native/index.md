# Re.Pack React Native APIs

## Table of contents

### Classes

- [Chunk](./classes/Chunk.md)
- [ChunkManager](./classes/ChunkManager.md)

### Interfaces

- [ChunkConfig](./interfaces/ChunkConfig.md)
- [ChunkManagerConfig](./interfaces/ChunkManagerConfig.md)
- [RemoteChunkLocation](./interfaces/RemoteChunkLocation.md)
- [StorageApi](./interfaces/StorageApi.md)

### Type aliases

- [RemoteChunkResolver](./types/RemoteChunkResolver.md)
