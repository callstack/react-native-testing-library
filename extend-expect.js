require('./build/matchers/extend-expect');
