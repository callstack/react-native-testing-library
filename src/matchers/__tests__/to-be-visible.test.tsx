import * as React from 'react';
import { View, Modal, Pressable } from 'react-native';
import { render } from '../..';
import '../extend-expect';

test('toBeVisible() on empty view', () => {
  const { getByTestId } = render(<View testID="test" />);
  expect(getByTestId('test')).toBeVisible();
});

test('toBeVisible() on view with opacity', () => {
  const { getByTestId } = render(
    <View testID="test" style={{ opacity: 0.2 }} />
  );
  expect(getByTestId('test')).toBeVisible();
});

test('toBeVisible() on view with 0 opacity', () => {
  const { getByTestId } = render(<View testID="test" style={{ opacity: 0 }} />);
  expect(getByTestId('test')).not.toBeVisible();
});

test('toBeVisible() on view with display "none"', () => {
  const { getByTestId } = render(
    <View testID="test" style={{ display: 'none' }} />
  );
  expect(
    getByTestId('test', { includeHiddenElements: true })
  ).not.toBeVisible();
});

test('toBeVisible() on ancestor view with 0 opacity', () => {
  const { getByTestId } = render(
    <View style={{ opacity: 0 }}>
      <View>
        <View testID="test" />
      </View>
    </View>
  );
  expect(
    getByTestId('test', { includeHiddenElements: true })
  ).not.toBeVisible();
});

test('toBeVisible() on ancestor view with display "none"', () => {
  const { getByTestId } = render(
    <View style={{ display: 'none' }}>
      <View>
        <View testID="test" />
      </View>
    </View>
  );
  expect(
    getByTestId('test', { includeHiddenElements: true })
  ).not.toBeVisible();
});

test('toBeVisible() on empty Modal', () => {
  const { getByTestId } = render(<Modal testID="test" />);
  expect(getByTestId('test')).toBeVisible();
});

test('toBeVisible() on view within modal', () => {
  const { getByTestId } = render(
    <Modal>
      <View>
        <View testID="view-within-modal" />
      </View>
    </Modal>
  );
  expect(getByTestId('view-within-modal')).toBeVisible();
});

test('toBeVisible() on view within not visible modal', () => {
  const { getByTestId, queryByTestId } = render(
    <Modal testID="test" visible={false}>
      <View>
        <View testID="view-within-modal" />
      </View>
    </Modal>
  );

  expect(getByTestId('test')).not.toBeVisible();
  // Children elements of not visible modals are not rendered.
  expect(() =>
    expect(getByTestId('view-within-modal')).not.toBeVisible()
  ).toThrow();
  expect(queryByTestId('view-within-modal')).toBeNull();
});

test('toBeVisible() on not visible modal', () => {
  const { getByTestId } = render(<Modal testID="test" visible={false} />);
  expect(
    getByTestId('test', { includeHiddenElements: true })
  ).not.toBeVisible();
});

test('toBeVisible() on inaccessible view', () => {
  const { getByTestId, update } = render(
    <View testID="test" aria-hidden={true} />
  );
  expect(
    getByTestId('test', { includeHiddenElements: true })
  ).not.toBeVisible();

  update(<View testID="test" />);
  expect(getByTestId('test')).toBeVisible();
});

test('toBeVisible() on view within inaccessible view', () => {
  const { getByTestId } = render(
    <View aria-hidden={true}>
      <View>
        <View testID="test" />
      </View>
    </View>
  );
  expect(
    getByTestId('test', { includeHiddenElements: true })
  ).not.toBeVisible();
});

test('toBeVisible() on inaccessible view (iOS)', () => {
  const { getByTestId, update } = render(
    <View testID="test" accessibilityElementsHidden />
  );
  expect(
    getByTestId('test', { includeHiddenElements: true })
  ).not.toBeVisible();

  update(<View testID="test" accessibilityElementsHidden={false} />);
  expect(getByTestId('test')).toBeVisible();
});

test('toBeVisible() on view within inaccessible view (iOS)', () => {
  const { getByTestId } = render(
    <View accessibilityElementsHidden>
      <View>
        <View testID="test" />
      </View>
    </View>
  );
  expect(
    getByTestId('test', { includeHiddenElements: true })
  ).not.toBeVisible();
});

test('toBeVisible() on inaccessible view (Android)', () => {
  const { getByTestId, update } = render(
    <View testID="test" importantForAccessibility="no-hide-descendants" />
  );
  expect(
    getByTestId('test', { includeHiddenElements: true })
  ).not.toBeVisible();

  update(<View testID="test" importantForAccessibility="auto" />);
  expect(getByTestId('test')).toBeVisible();
});

test('toBeVisible() on view within inaccessible view (Android)', () => {
  const { getByTestId } = render(
    <View importantForAccessibility="no-hide-descendants">
      <View>
        <View testID="test" />
      </View>
    </View>
  );
  expect(
    getByTestId('test', { includeHiddenElements: true })
  ).not.toBeVisible();
});

test('toBeVisible() on null elements', () => {
  expect(() => expect(null).toBeVisible()).toThrowErrorMatchingInlineSnapshot(`
      "expect(received).toBeVisible()

      received value must be a host element.
      Received has value: null"
    `);
});

test('toBeVisible() on non-React elements', () => {
  expect(() =>
    expect({ name: 'Non-React element' }).not.toBeVisible()
  ).toThrow();
  expect(() => expect(true).not.toBeVisible()).toThrow();
});

test('toBeVisible() throws an error when expectation is not matched', () => {
  const { getByTestId, update } = render(<View testID="test" />);
  expect(() =>
    expect(getByTestId('test')).not.toBeVisible()
  ).toThrowErrorMatchingSnapshot();

  update(<View testID="test" style={{ opacity: 0 }} />);
  expect(() =>
    expect(getByTestId('test')).toBeVisible()
  ).toThrowErrorMatchingSnapshot();
});

test('toBeVisible() on Pressable with function style prop', () => {
  const { getByTestId } = render(
    <Pressable testID="test" style={() => ({ backgroundColor: 'blue' })} />
  );
  expect(getByTestId('test')).toBeVisible();
});
