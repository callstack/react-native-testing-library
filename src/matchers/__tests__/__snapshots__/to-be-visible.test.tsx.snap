// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeVisible() throws an error when expectation is not matched 1`] = `
"expect(element).not.toBeVisible()

Received element is visible:
  <View
    testID="test"
  />"
`;

exports[`toBeVisible() throws an error when expectation is not matched 2`] = `
"expect(element).toBeVisible()

Received element is not visible:
  <View
    style={
      {
        "opacity": 0,
      }
    }
    testID="test"
  />"
`;
